cypress-tests/cypress/support/commands.js

```diff
@@ -722,19 +722,61 @@ Cypress.Commands.add("handleRedirection", (globalState, expected_redirection) =>
             cy.get('#test-source-authorize-3ds').click();
           })
       })
-  } else {
-    cy.wait(10000);
+  } 
+  else if (globalState.get("connectorId") == "adyen") {
+    cy.get('iframe')
+      .its('0.contentDocument.body')
+      .within((body) => {
+        cy.get('input[type="password"]').click();
+        cy.get('input[type="password"]').type("password");
+        cy.get('#buttonSubmit').click();
+      })
+  } 
+  else if (globalState.get("connectorId") === "cybersource" || globalState.get("connectorId") === "bankofamerica" ) {
+    cy.get('iframe')
+      .its('0.contentDocument.body')
+      .within((body) => {
+        cy.get('input[type="text"]').click().type("1234");
+        cy.get('input[value="SUBMIT"]').click();
+  })
   }
-
+  else if (globalState.get("connectorId") === "trustpay" ) {
+    cy.get('form[name="challengeForm"]',{ timeout: 10000 }).should('exist').then(form => {
+        cy.get('#outcomeSelect').select('Approve').should('have.value', 'Y')
+        cy.get('button[type="submit"]').click();
+  })
+  }
+  else if (globalState.get("connectorId") === "nmi" || globalState.get("connectorId") === "noon") {
+    cy.get('iframe',{ timeout: 100000 })
+      .its('0.contentDocument.body')
+      .within((body) => {
+        cy.get('iframe',{ timeout: 10000 })
+          .its('0.contentDocument.body')
+          .within((body) => {
+        cy.get('form[name="cardholderInput"]',{ timeout: 10000 }).should('exist').then(form => {
+          cy.get('input[name="challengeDataEntry"]').click().type("1234");
+          cy.get('input[value="SUBMIT"]').click();
+          })
+      })
+    })
+}
+  
```

Comment:
Also, can we have this if else block to be in alphabetical order of connector names for easier look ups? We will anyway move this to a separate file so that we can have all redirection in one place.

Reply:
Sure I made the change